app/grandchallenge/utilization/migrations/0003_auto_20250520_1012.py

```diff
@@ -0,0 +1,121 @@
+from django.db import migrations
+from django.db.models import F
+
+
+def create_evaluation_utilizations(apps, schema_editor):
+    Evaluation = apps.get_model("evaluation", "Evaluation")  # noqa: N806
+    EvaluationUtilization = apps.get_model(  # noqa: N806
+        "utilization", "EvaluationUtilization"
+    )
+    for evaluation in (
+        Evaluation.objects.annotate(
+            duration=F("completed_at") - F("started_at")
+        )
+        .filter(evaluation_utilization__isnull=True)
+        .iterator()
+    ):
+        kwargs = dict(
+            created=evaluation.created,
+            duration=evaluation.duration,
+            compute_cost_euro_millicents=evaluation.compute_cost_euro_millicents,
+            external_evaluation=evaluation.submission.phase.external_evaluation,
+            algorithm_image=evaluation.submission.algorithm_image,
+            archive=evaluation.submission.phase.archive,
+            challenge=evaluation.submission.phase.challenge,
+            creator=evaluation.submission.creator,
+            evaluation=evaluation,
+            phase=evaluation.submission.phase,
+        )
+        if evaluation.submission.algorithm_image is not None:
+            kwargs.update(
+                algorithm=evaluation.submission.algorithm_image.algorithm,
+            )
+        EvaluationUtilization.objects.create(**kwargs)
+
+
+def create_job_utilizations(apps, schema_editor):
+    Job = apps.get_model("algorithms", "Job")  # noqa: N806
+    JobUtilization = apps.get_model(  # noqa: N806
+        "utilization", "JobUtilization"
+    )
+    for job in (
+        Job.objects.annotate(duration=F("completed_at") - F("started_at"))
+        .filter(job_utilization__isnull=True)
+        .iterator()
+    ):
+        JobUtilization.objects.create(
+            created=job.created,
+            duration=job.duration,
+            compute_cost_euro_millicents=job.compute_cost_euro_millicents,
+            algorithm=job.algorithm_image.algorithm,
+            algorithm_image=job.algorithm_image,
+            creator=job.creator,
+            job=job,
+        )
+
+
+def set_challenges_to_job_utilizations(apps, schema_editor):
+    Job = apps.get_model("algorithms", "Job")  # noqa: N806
+    Permission = apps.get_model("auth", "Permission")  # noqa: N806
+    Challenge = apps.get_model("challenges", "Challenge")  # noqa: N806
+    challenges = Challenge.objects.all()
+    if challenges.exists():
+        permission = Permission.objects.get(
+            codename="view_job",
+            content_type__app_label="algorithms",
+            content_type__model="job",
+        )
+        for challenge in challenges:
+            jobs = Job.objects.filter(
+                jobgroupobjectpermission__group=challenge.admins_group,
+                jobgroupobjectpermission__permission=permission,
+            ).distinct()
+            for job in jobs:
+                job.update_utilization(challenge=challenge)
```

Comment:
Has this been tested? I do not think that you have access to custom methods here. Also check the iteration and memory requirements. It may be possible to replace this with a batch update https://docs.djangoproject.com/en/5.2/ref/models/querysets/#update